Core/OfficeDevPnP.Core/Extensions/StreamExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OfficeDevPnP.Core.Extensions
+{
+    public static class StreamExtensions
+    {
+        public static MemoryStream ToMemoryStream(this Stream source)
+        {
+            var stream = source as MemoryStream;
+            if (stream != null) return stream;
+            MemoryStream target = new MemoryStream();
+            const int bufSize = 65535;
+            byte[] buf = new byte[bufSize];
+            int bytesRead = -1;
+            while ((bytesRead = source.Read(buf, 0, bufSize)) > 0)
+                target.Write(buf, 0, bytesRead);
+            return target;
+        }
```

Comment:
Hi Mikael, 
Here I would set the Position of the resulting MemoryStream to 0, because most likely the incoming source stream is positioned at 0 and the resulting one should have the same position. Just to avoid exceptions when processing the stream content, later on.